### ISSUE TITLE
Provide support for explicitly configured bundle parts

### ISSUE THREAD

**Issue by ledina** [IBM/cics-bundle-gradle#14]
Allow name override for WAR and EAR (not supported for OSGi).
Clarification: You can override the `name` of an OSGi bundle but not the `symbolicName`.
Should be able to specify jvmServer to override the defaultJvmServer value.

**Comment by tom-foyle** [comment]
Proposed solution: Add a new extension block `cicsBundleOverrides` in which multiple `override` sub-blocks can be added to override the name, jvmserver, and file extension (i.e. bundle part type) of a particular bundle part. Specify which bundle part to override by supplying it's artifact filename.
```
dependencies {
    cicsBundle(group: 'org.codehaus.cargo', name: 'simple-bundle', version: '1.7.7', ext: 'jar')
    cicsBundle(group: 'org.codehaus.cargo', name: 'simple-war', version: '1.7.7', ext: 'war')
}

cicsBundleOverrides {
    // Override name and jvmserver of simple-bundle-1.7.7.
    override {
        filename = 'simple-bundle-1.7.7.jar'
        name = 'new-name'
        jvmserver = 'NEWJVMS'
    }
    // Override file extension of simple-war-1.7.7 so it is processed as an ear bundle part instead.
    override {
        filename = 'simple-war-1.7.7.war'
        extension = 'ear'
    }
}
```

**Comment by tom-foyle** [comment]
Proposed solution 2: Adapt the extension so that the extra config can go inline with the dependency declaration. This feels more natural and means that the user doesn't need to know what the resolved filename will be.

Two types of notations can be used. Either specify the 'bundlePart' config using a closure:
'''
dependencies {
    bundleParts {
        // Override name and jvmserver of simple-bundle-1.7.7.
        bundlePart {
            dependency = cicsBundle(group: 'org.codehaus.cargo', name: 'simple-bundle', version: '1.7.7', ext: 'jar')
            name = 'new-name'
            jvmserver = 'NEWJVMS'
        }
        // Override file extension of simple-war-1.7.7 so it is processed as an ear instead.
        bundlePart {
            dependency = cicsBundle(group: 'org.codehaus.cargo', name: 'simple-war', version: '1.7.7', ext: 'war')
            extension = 'ear'
        }
}
'''
Or on a single line using a method which takes a map as the argument:
'''
dependencies {
    bundleParts {
        // Override file extension of simple-war-1.7.7 so it is processed as an ear instead.
        bundlePart (dependency: cicsBundle(group: 'org.codehaus.cargo', name: 'simple-war', version: '1.7.7', ext: 'war'), extension: 'ear')
}
'''

**Comment by tom-foyle** [comment]
Proposed solution 3: Use strong types like `cicsBundleWar` rather than `bundlePart`. Use DSL methods to allow bundle parts to be specified at the top level inside the `dependencies` block so they no longer need to be nested under a `bundleParts` parent extension.
```
dependencies {
    // Normal dependency with no overrides.
    cicsBundle 'org.codehaus.cargo:simple-ear:1.7.7@ear'
    // Override name and jvmserver of simple-bundle-1.7.7.
    cicsBundleOsgi {
        dependency = cicsBundle(group: 'org.codehaus.cargo', name: 'simple-bundle', version: '1.7.7', ext: 'jar')
        name = 'new-name'
        jvmserver = 'NEWJVMS'
    }
    // Override file extension of simple-war-1.7.7 so it is processed as an ear instead.
    cicsBundleEar(dependency: cicsBundle('org.codehaus.cargo:simple-war:1.7.7@war'))
}
```

**Comment by ledina** [comment]
Resolved by #63
